node/processingOnlyNode/coreComponents.go

```diff
@@ -0,0 +1,381 @@
+				},
+			},
+		},
+		ConfigDir:          "",
+		EpochNotifier:      instance.epochNotifier,
+		WasmVMChangeLocker: instance.wasmVMChangeLocker,
+	}
+	gasScheduleNotifier, err := forking.NewGasScheduleNotifier(argsGasSchedule)
+	if err != nil {
+		return nil, err
+	}
+
+	builtInCostHandler, err := economics.NewBuiltInFunctionsCost(&economics.ArgsBuiltInFunctionCost{
+		ArgsParser:  smartContract.NewArgumentParser(),
+		GasSchedule: gasScheduleNotifier,
+	})
+	if err != nil {
+		return nil, err
+	}
+
+	argsEconomicsHandler := economics.ArgsNewEconomicsData{
+		TxVersionChecker:            instance.txVersionChecker,
+		BuiltInFunctionsCostHandler: builtInCostHandler,
+		Economics:                   &args.EconomicsConfig,
+		EpochNotifier:               instance.epochNotifier,
+		EnableEpochsHandler:         instance.enableEpochsHandler,
+	}
+
+	instance.economicsData, err = economics.NewEconomicsData(argsEconomicsHandler)
+	if err != nil {
+		return nil, err
+	}
+	instance.apiEconomicsData = instance.economicsData
+
+	// TODO check if we need this
+	instance.ratingsData = nil
+	instance.rater = nil
+
+	instance.genesisNodesSetup, err = sharding.NewNodesSetup(args.NodesSetupPath, instance.addressPubKeyConverter, instance.validatorPubKeyConverter, args.NumShards)
+	if err != nil {
+		return nil, err
+	}
+	// TODO check if we need nodes shuffler
+	instance.nodesShuffler = nil
+
+	instance.roundNotifier = forking.NewGenericRoundNotifier()
+	instance.enableRoundsHandler, err = enablers.NewEnableRoundsHandler(args.RoundsConfig, instance.roundNotifier)
+	if err != nil {
+		return nil, err
+	}
+
+	instance.epochStartNotifierWithConfirm = notifier.NewEpochStartSubscriptionHandler()
+	instance.chanStopNodeProcess = args.ChanStopNodeProcess
+	instance.genesisTime = time.Unix(instance.genesisNodesSetup.GetStartTime(), 0)
+	instance.chainID = args.Cfg.GeneralSettings.ChainID
+	instance.minTransactionVersion = args.Cfg.GeneralSettings.MinTransactionVersion
+	instance.encodedAddressLen, err = computeEncodedAddressLen(instance.addressPubKeyConverter)
+	if err != nil {
+		return nil, err
+	}
+
+	instance.nodeTypeProvider = nodetype.NewNodeTypeProvider(core.NodeTypeObserver)
+	instance.processStatusHandler = statusHandler.NewProcessStatusHandler()
+
+	pubKeyBytes, err := instance.validatorPubKeyConverter.Decode(args.Cfg.Hardfork.PublicKeyToListenFrom)
+	if err != nil {
+		return nil, err
+	}
+	instance.hardforkTriggerPubKey = pubKeyBytes
+
+	return instance, nil
+}
+
+func computeEncodedAddressLen(converter core.PubkeyConverter) (uint32, error) {
+	emptyAddress := bytes.Repeat([]byte{0}, converter.Len())
+	encodedEmptyAddress, err := converter.Encode(emptyAddress)
+	if err != nil {
+		return 0, err
+	}
+
+	return uint32(len(encodedEmptyAddress)), nil
+}
+
+func (c *coreComponentsHolder) InternalMarshalizer() marshal.Marshalizer {
+	return c.internalMarshaller
+}
+
+func (c *coreComponentsHolder) SetInternalMarshalizer(marshalizer marshal.Marshalizer) error {
```

Comment:
do we need this set?

Reply:
no sure I just implemented the methods from the `factory.CoreComponentsHolder`